src/V3LinkLValue.cpp

```diff
@@ -205,6 +205,21 @@ class LinkLValueVisitor : public AstNVisitor {
         }
         m_setRefLvalue = last_setRefLvalue;
     }
+    void prepost_visit(AstNodeTriop* nodep) {
+        bool last_setRefLvalue = m_setRefLvalue;
+        {
+            m_setRefLvalue = true;
+            iterateAndNextNull(nodep->lhsp());
+            iterateAndNextNull(nodep->thsp());
+            m_setRefLvalue = false;
+            iterateAndNextNull(nodep->rhsp());
```

Comment:
I can see how "shuffle" was very non-obvious!  I mean do it this way:  Then see matching change in verilog.y and your visitor.

```suggestion
            m_setRefLvalue = false;
            iterateAndNextNull(nodep->lhsp());
            iterateAndNextNull(nodep->thsp());
            m_setRefLvalue = true;
            iterateAndNextNull(nodep->rhsp());
```